runtime/parachains/src/hrmp.rs

```diff
@@ -1167,6 +1237,35 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+impl<T: Config> Pallet<T> {
+	/// Decreases the open channel request count for the given sender. If the value reaches zero
+	/// it is removed completely.
+	fn decrease_open_channel_request_count(sender: ParaId) {
+		let new_open_channel_req_cnt =
+			<Self as Store>::HrmpOpenChannelRequestCount::get(&sender).saturating_sub(1);
+		if new_open_channel_req_cnt != 0 {
+			<Self as Store>::HrmpOpenChannelRequestCount::insert(&sender, new_open_channel_req_cnt);
+		} else {
+			<Self as Store>::HrmpOpenChannelRequestCount::remove(&sender);
+		}
```

Comment:
this could be done a bit more cleanly with a mutate_exists function i think.

Reply:
Thanks for the suggestion!